### ISSUE TITLE
[MC class] Sign of the first configuration's weight

### ISSUE THREAD

**Issue by amoutenet** [TRIQS/triqs#532]
Dear all,

I open an issue to go back to a discussion we had several times: when defining your MC class, you need the sign of the first configuration's weight (which already seems strange to me as it should not be used to construct the class, should it?).
However, the construction of the MC class provides the random number generator that I (we?) use throughout the code to change configurations. If you do not want to begin from a specific configuration that might be a bit singular, you therefore need this RNG to construct it... and therefore you do not know its weight's sign before having the RNG!

I think it would be much cleaner to have the first weight in the warmup_and_accumulate function, as it is here that it is used. The actual construction makes me define a different RNG and construct me configuration before creating the MC class, which is a bit confusing in the solver_core file.

Best,
Alice

**Comment by amoutenet** [comment]
Hi everyone,
This issue has been around for a while and it's getting back to me as I don't know what will be the sign of the fist config before defining my MC class. I can work around it, but would there be a way to move it to warmup_and_accumulate or at least to change its value after the initialization of the MC?
Thanks!
Alice

**Comment by amoutenet** [comment]
@parcollet

**Comment by Wentzell** [comment]
This is fixed in ab1244ae
We deprecated the old constructor and you can now pass the initial sign to `warmup_and_accumulate`

```c++
int warmup_and_accumulate(uint64_t n_warmup_cycles, uint64_t n_accumulation_cycles, uint64_t length_cycle, std::function<bool()> stop_callback, MCSignType sign_init = 1)
```
